src/shmem_node_util.c

```diff
@@ -0,0 +1,92 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "shmem_internal.h"
+#include "shmem_node_util.h"
+
+
+char *shmem_internal_location_array = NULL;
+
+static int node_util_is_initialized = 0;
+
+
+int shmem_node_util_init(void) {
+    int ret;
+
+    if (!node_util_is_initialized) {
+        shmem_internal_location_array = malloc(sizeof(char) * shmem_internal_num_pes);
+        if (NULL == shmem_internal_location_array) {
+            RAISE_ERROR_STR("Error: out of memory when allocating node_util location array");
+            return 1;
```

Comment:
The return is unreachable since `RAISE_ERROR_*` will abort.  Do you want `RETURN_ERROR_*` instead?  If not, should this routine be define to return a value?